src/base/rss/rss_autodownloader.cpp

```diff
@@ -266,6 +270,38 @@ void AutoDownloader::importRulesFromLegacyFormat(const QByteArray &data)
         insertRule(AutoDownloadRule::fromLegacyDict(val.toHash()));
 }
 
+QStringList AutoDownloader::getSmartEpisodeFilters() const
+{
+    QVariant filtersSetting = SettingsStorage::instance()->loadValue(SettingsKey_SmartEpisodeFilter);
+
+    if (filtersSetting.isNull())
+    {
+        QStringList filters;
+        filters.append("(?:s(\\d+)e(\\d+))"); //Format 1: s01e01
+        filters.append("(?:(\\d+)x(\\d+))");  //Format 2: 01x01
+        filters.append("(\\d{4}[.\\-]\\d{1,2}[.\\-]\\d{1,2})"); //Format 3: 2017.01.01
+        filters.append("(\\d{1,2}[.\\-]\\d{1,2}[.\\-]\\d{4})"); //Format 4: 01.01.2017
+        return filters;
+    }
+
+    return filtersSetting.toStringList();
+}
+
+QRegularExpression AutoDownloader::getSmartEpisodeRegex() const
+{
+    return this->m_smartEpisodeRegex;
+}
+
+void AutoDownloader::setSmartEpisodeFilters(const QStringList &filters)
+{
+    SettingsStorage::instance()->storeValue(SettingsKey_SmartEpisodeFilter, filters);
+
+    QString regex = QString("(?:^|[^a-z0-9])(?:%1)(?:[^a-z0-9]|$)").arg(filters.join('|'));
```

Comment:
Can you explain this regex, please.

Reply:
This regex combines all of the smart episode filters the user enters in the settings, and searches for them between non-alphanumeric characters, or the start/end of the article title.

Reply:
I asked because I doubted that this regex is exactly what you want... And I was right.
>searches for them between non-alphanumeric characters, or the start/end of the article title

It searches between non-(latin lower-case letters or numbers) characters only. I don't think this behavior is consistent with the public use of this feature.
What about `\b(?:%1)\b`?

Reply:
One small point, the regex is created as case-insensitive, so it does get upper case as well, but yes your point stands, it matches latin characters only. \b isn't a good choice, and is something I looked at early on, since it includes the '_' character which is something I would want to be a boundary. Also, \b is just boundaries between [A-Za-z0-9_], which is only one character different to what I've got.

If I had examples of non-latin based articles I could make a better choice, but I feel this is fine for a first attempt.

Reply:
>it includes the '' character which is something I would want to be a boundary

Unfortunately, I don't understand this.
>Also, \b is just boundaries between [A-Za-z0-9], which is only one character different to what I've got.

\b is boundary between any "word" character (i.e. \w) and non-"word" character (including start/end of line). When you enable Unicode support in QRegularExpression \w includes also non-latin alphanumeric characters.

Reply:
The main disadvantage is \w includes underscore so it will not match, e.g., "Some_series_name_s01e03". `(?<=^|[\W_])s(\d+)e(\d+)(?=[\W_]|$)` should match.

Reply:
I'd like to suggest: `(?:^|[\W_])(?:%1)(?:[\W_]|$)`

That should match the pattern between any word boundary characters as well as underscore. Be sure to include `QRegularExpression::UseUnicodePropertiesOption` when constructing the `QRegularExpression` so that `\W` will properly match unicode word boundaries.

Reply:
@briankendall, your suggestion is almost the same as mine above (except I have example only for some particular format).

Reply:
@glassez I thought `\b` matches a backspace character?

Reply:
> Unfortunately, I don't understand this.
There was meant to be an underscore between the quotes, but I had forgotten that was a markdown character.

I've played around with it a bit, and settled on `(?:[_]|\b)`, with Unicode support enabled. `\b` isn't a character class so you can't place it inside [ ].

Reply:
My last example I talk about isn't contain `\b`:
`(?<=^|[\W_])s(\d+)e(\d+)(?=[\W_]|$)`
>Using `[\b_]` and `[\W_]` are probably equivalent.

IIRC, `\b` doesn't work in this case (at least, as you expect).

Reply:
>I've played around with it a bit, and settled on (?:[_]|\b), with Unicode support enabled. \b isn't a character class so you can't place it inside [ ].

`(?<=^|[\W_])(?:%1)(?=[\W_]|$)` is what you need. Do try it on https://regexr.com/ (don't forget to select PCRE as RegEx Engine).

P.S. `[_]` is a perversion, you don't need `[]` for a single character.